### DISCUSSION TITLE
Is it possible to read and write files?

### DISCUSSION THREAD

**Discussion by ghost** [lcn2/calc#102] [Q&A] [answered]
I mean files in general, not `calc` modules.

**Comment by lcn2** [comment, marked as answer]
Yes, calc can open, read and write files.  

For a list of reading, writing, open and I/O support functions see [discussion comment 7180636](https://github.com/lcn2/calc/discussions/102#discussioncomment-7180636).

For an example, see [discussion comment 6770367](https://github.com/lcn2/calc/discussions/97#discussioncomment-6770367).

Does the above answer your question?

**Reply by lcn2** [reply]
Calc has a full set of file related built in functions.

On the read side, in addition to the usual functions:

* [fgetc](https://github.com/lcn2/calc/blob/master/help/fgetc)
* [fgets](https://github.com/lcn2/calc/blob/master/help/fgets)
* [scanf](https://github.com/lcn2/calc/blob/master/help/scanf)
* [fscanf](https://github.com/lcn2/calc/blob/master/help/fscanf)

calc has some special built in read functions such as:

* [fgetline](https://github.com/lcn2/calc/blob/master/help/fgetline)
* [fgetfield](https://github.com/lcn2/calc/blob/master/help/fgetfield)
* [fgetstr](https://github.com/lcn2/calc/blob/master/help/fgetstr)
* [fgetfile](https://github.com/lcn2/calc/blob/master/help/fgetfile)
* [fscan](https://github.com/lcn2/calc/blob/master/help/fscan)

On the write side, in addition to the usual functions:

* [fprintf](https://github.com/lcn2/calc/blob/master/help/fprintf)
* [fputc](https://github.com/lcn2/calc/blob/master/help/fputc)
* [fputs](https://github.com/lcn2/calc/blob/master/help/fputs)

calc has some special built in write functions such as:

* [fputstr](https://github.com/lcn2/calc/blob/master/help/fputstr)
* [prompt](https://github.com/lcn2/calc/blob/master/help/prompt)

There are calc built in functions to open files:

* [fopen](https://github.com/lcn2/calc/blob/master/help/fopen)
* [freopen](https://github.com/lcn2/calc/blob/master/help/freopen)
* [fpathopen](https://github.com/lcn2/calc/blob/master/help/fpathopen)

As well as the usual calc built in I/O support functions:

* [fseek](https://github.com/lcn2/calc/blob/master/help/fseek)
* [ftell](https://github.com/lcn2/calc/blob/master/help/ftell)
* [fflush](https://github.com/lcn2/calc/blob/master/help/fflush)
* [fclose](https://github.com/lcn2/calc/blob/master/help/fclose)
* [feof](https://github.com/lcn2/calc/blob/master/help/feof)
* [ferror](https://github.com/lcn2/calc/blob/master/help/ferror)
* [files](https://github.com/lcn2/calc/blob/master/help/files)
* [fsize](https://github.com/lcn2/calc/blob/master/help/fsize)
* [isfile](https://github.com/lcn2/calc/blob/master/help/isfile)

And there is more.

For more information see:

```
; help file
```

And:

```
SEE ALSO
    errno, fclose, feof, ferror, fflush, fgetc, fgetline, fgets, files, fopen,
    fprintf, fputc, fputs, fseek, fsize, ftell, isfile, printf, prompt
```

See also See [discussion comment 6770367](https://github.com/lcn2/calc/discussions/97#discussioncomment-6770367).
